applications/ShapeOptimizationApplication/python_scripts/analyzer_factory.py

```diff
@@ -193,57 +194,82 @@ def __RequestResponsesAccordingDependencies(self, communicator):
             if len(dependencies) > 0:
                 if communicator.isRequestingValueOf(response_id):
-                    combined_value = self.__ComputeCombinedValuesRecursively(dependencies, communicator)
+                    combined_value = self.__ComputeCombinedValuesRecursively(dependencies, combination_type, communicator)
                     communicator.reportValue(response_id, combined_value)
 
-        for response_id, dependencies, _ in self.dependency_graph:
+        for response_id, _, dependencies, _ in self.dependency_graph:
             if len(dependencies) > 0:
                 if communicator.isRequestingGradientOf(response_id):
                     combined_gradient = self.__ComputeCombinedGradientsRecursively(dependencies, communicator)
                     communicator.reportGradient(response_id, combined_gradient)
 
     # --------------------------------------------------------------------------
-    def __ComputeCombinedValuesRecursively(self, dependencies, communicator):
-        combined_value = 0.0
+    def __ComputeCombinedValuesRecursively(self, dependencies, combination_type, communicator):
+        combined_value = None
+        relevant_responses = []
 
-        for response_id, dependencies, weight in dependencies:
-            if len(dependencies) > 0:
-                value = self.__ComputeCombinedValuesRecursively(dependencies, communicator)
+        for response_id, weight, sub_dependencies, sub_combination_type in dependencies:
+            if len(sub_dependencies) > 0:
+                value = self.__ComputeCombinedValuesRecursively(sub_dependencies, sub_combination_type, communicator)
                 communicator.reportValue(response_id, value)
             else:
                 value = communicator.getStandardizedValue(response_id)
-            combined_value += weight*value
+
+            value = weight*value
+
+            if combined_value is None:
+                combined_value = value
+                relevant_responses.append(response_id)
+            else:
+                if combination_type == "sum":
+                    combined_value += value
+                    relevant_responses.append(response_id)
+                elif combination_type == "max":
+                    if value > combined_value:
+                        relevant_responses[0] = response_id
+                        combined_value = value
+                elif combination_type == "min":
+                    if value < combined_value:
+                        relevant_responses[0] = response_id
+                        combined_value = value
+                else:
+                    raise NameError("The following combination_type is not supported:" + combination_type)
+
+        for response_id, _, _, _ in dependencies:
+            if response_id not in relevant_responses:
+                communicator.initializeRequest(response_id, value_request=False)
```

Comment:
```suggestion
                communicator.UpdateRequest(response_id, value_request=True, gradient_request=False)
```
To my understanding at this stage you have already used the value, so it should "remain" requested, but you can tell now that the gradient is not needed anymore?
Or are the requests set to false when the corresponding values are set at the communicator?

Reply:
Anyway i would prefer to write both keyword arguments here

Reply:
This shows how kwargs can be enforced: https://stackoverflow.com/questions/2965271/forced-naming-of-parameters-in-python

Reply:
I adjusted it to use kwargs. Totally forgot about this feature. Thanks for the hint. Also, I used your suggested function name as it expresses the intent much better.

In fact, it is an updating of the requests within the analyzer. Also, just the gradient request ist updated right after combining the values (determining the max value).